charts/istio-endpoints/README.md

```diff
@@ -0,0 +1,112 @@
+# istio-endpoints
+
+Per-Namespace Istio Configuration Chart
+
+![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: latest](https://img.shields.io/badge/AppVersion-latest-informational?style=flat-square)
+
+[elasticache]: https://aws.amazon.com/elasticache/
+[serviceentry]: https://istio.io/latest/docs/reference/config/networking/service-entry/
+[envoyfilter]: https://istio.io/latest/docs/reference/config/networking/envoy-filter/
+[sidecar]: https://istio.io/latest/docs/reference/config/networking/sidecar/
+[envoy]: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/other_protocols/redis
+
+This small chart helps configure the [Istio Proxy Sidecar][sidecar] pods in a
+given `Namespace` to route traffic to AWS managed [ElastiCache][elasticache]
+clusters or services. This assumes that you are running the Istio Service Mesh
+inside of your cluster already, and have a need to use [Envoy's
+RedisProxy][envoy] configuration to handle traffic routing to the ElastiCache
+services.
+
+**Pre-Req: `PILOT_ENABLE_REDIS_FILTER=true` must be set in your Istio Pilot Config**
+
+If you are using the [Istio
+Operator](https://istio.io/latest/docs/reference/commands/operator/), your
+`IstioOperator` resource must enable the `RedisProxy` filter. This filter
+allows Envoy to understand the `REDIS` protocol, and automatically configures
+the filter when necessary.
+
+```yaml
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: istio-control-plane
+spec:
+  components:
+    pilot:
+      k8s:
+        env:
+          - name: PILOT_ENABLE_REDIS_FILTER
+            value: 'true'
+```
+
+## Usage
+
+### Add to your `Chart.yaml` dependencies:
+
+```diff
+  apiVersion: v2
+  name: simple-app
+  description: Default Microservice Helm Chart
+  type: application
+  version: 0.7.0
+  appVersion: latest
++ dependencies:
++   - name: istio-endpoints
++     repository: https://k8s-charts.nextdoor.com
++     version: 0.1.0
+  maintainers:
+    - name: diranged
+      email: matt@nextdoor.com
+```
+
+### Set up your list of ElastiCache AWS Endpoints
+
+```yaml
+# values.yaml
+istio-endpoints:
+  test:
+    address: staging-cluster.abcd8x.clustercfg.usw2.cache.amazonaws.com
+    bindPort: 1234          # required
```

Comment:
how do I reference isto-endpoints.test.address in my application `{{ .Values.istio-endpoints.test.address }}`? or do I have to go the `global` thing?

Reply:
Your application actually points to the `bindAddress` and `bindPort` that you have configured here, so eg: `127.0.0.2:6379`.